ngraph/core/src/op/non_max_suppression.cpp

```diff
@@ -530,3 +533,593 @@ void op::v4::NonMaxSuppression::validate_and_infer_types()
+static constexpr size_t scores_port = 1;
+static constexpr size_t iou_threshold_port = 3;
+static constexpr size_t score_threshold_port = 4;
+static constexpr size_t soft_nms_sigma_port = 5;
+
+float op::v5::NonMaxSuppression::iou_threshold_from_input() const
+{
+    float iou_threshold = 0.0f;
+
+    const auto iou_threshold_input =
+        as_type_ptr<op::Constant>(input_value(iou_threshold_port).get_node_shared_ptr());
+    iou_threshold = iou_threshold_input->cast_vector<float>().at(0);
+
+    return iou_threshold;
+}
+
+float op::v5::NonMaxSuppression::score_threshold_from_input() const
+{
+    float score_threshold = 0.0f;
+
+    const auto score_threshold_input =
+        as_type_ptr<op::Constant>(input_value(score_threshold_port).get_node_shared_ptr());
+    score_threshold = score_threshold_input->cast_vector<float>().at(0);
+
+    return score_threshold;
+}
+
+float op::v5::NonMaxSuppression::soft_nms_sigma_from_input() const
+{
+    float soft_nms_sigma = 0.0f;
+
+    const auto soft_nms_sigma_input =
+        as_type_ptr<op::Constant>(input_value(soft_nms_sigma_port).get_node_shared_ptr());
+    soft_nms_sigma = soft_nms_sigma_input->cast_vector<float>().at(0);
+
+    return soft_nms_sigma;
+}
+
+bool ngraph::op::v5::NonMaxSuppression::visit_attributes(AttributeVisitor& visitor)
+{
+    visitor.on_attribute("box_encoding", m_box_encoding);
+    visitor.on_attribute("sort_result_descending", m_sort_result_descending);
+    visitor.on_attribute("output_type", m_output_type);
+    return true;
+}
+
+void op::v5::NonMaxSuppression::validate_and_infer_types()
+{
+    const auto boxes_ps = get_input_partial_shape(0);
+    const auto scores_ps = get_input_partial_shape(1);
+
+    // NonMaxSuppression produces triplets
+    // that have the following format: [batch_index, class_index, box_index]
+    PartialShape out_shape = {Dimension::dynamic(), 3};
+
+    validate();
+
+    set_output_type(0, m_output_type, out_shape);
+    set_output_type(1, element::f32, out_shape);
+    set_output_type(2, m_output_type, Shape{});
+}
+
+namespace ngraph
+{
+    template <>
+    EnumNames<op::v5::NonMaxSuppression::BoxEncodingType>&
+        EnumNames<op::v5::NonMaxSuppression::BoxEncodingType>::get()
+    {
+        static auto enum_names = EnumNames<op::v5::NonMaxSuppression::BoxEncodingType>(
+            "op::v5::NonMaxSuppression::BoxEncodingType",
+            {{"corner", op::v5::NonMaxSuppression::BoxEncodingType::CORNER},
+             {"center", op::v5::NonMaxSuppression::BoxEncodingType::CENTER}});
+        return enum_names;
+    }
+
+    constexpr DiscreteTypeInfo
+        AttributeAdapter<op::v5::NonMaxSuppression::BoxEncodingType>::type_info;
+
+    std::ostream& operator<<(std::ostream& s,
+                             const op::v5::NonMaxSuppression::BoxEncodingType& type)
+    {
+        return s << as_string(type);
+    }
+} // namespace ngraph
+
+static PartialShape infer_selected_indices_shape(const HostTensorVector& inputs,
```

Comment:
I propose to consolidate all static local functions and members in unnamed namespace.

Reply:
Done.